### DISCUSSION TITLE
VSCode Service Worker is overriding my Service Worker

### DISCUSSION THREAD

**Discussion by cemalgnlts** [CodinGame/monaco-vscode-api#312] [Q&A]
Hi,

I use a service worker, so I can present all the files offline.
*(https://github.com/cemalgnlts/jside/blob/main/public/sw.js#L49)*

But after the last update, probably because of the Welcome View service, it now registers a service worker in VSCode. Since VSCode's service worker does the fetch of the two files itself, I cannot serve them from cahce.

With the Developer Console, I printed the files accessed by both my Service Worker and the VSCode Service Worker to the console. As you can see in the picture, my service worker serves everything except an html file. Since I cannot serve this html file, it is not possible for the project to run completely offline.

https://github.com/microsoft/vscode/blob/a9561b8195c3f99ec7e8e72bc2307dcefced0d56/src/vs/workbench/contrib/webview/browser/pre/service-worker.js#L159

![Screenshot 2024-01-12 12 08 43](https://github.com/CodinGame/monaco-vscode-api/assets/45357531/c00f9d16-5ff0-41d2-84d0-a52ce2488740)

**Comment by CGNonofr** [comment]
Did you only try on localhost or on the deployed version of you app? it seems there is a [condition for localhost](https://github.com/microsoft/vscode/blob/a9561b8195c3f99ec7e8e72bc2307dcefced0d56/src/vs/workbench/contrib/webview/browser/pre/service-worker.js#L203) and that's the only place a fetch is done directly

Otherwise, it seems that the worker delegate the request back to the main page (but I may be wrong as I'm not a SW expert :) )

**Reply by cemalgnlts** [reply]
I'm trying the deployed version. Yes, you are right, none of the conditions fit. But then the request is not caught by my service worker, it is left to the browser. I'll look into it some more.

**Reply by cemalgnlts** [reply]
Is there a need for this service worker? I have disabled it and everything seems normal. The conditions never seem to be met and it looks like (vscode) Service Worker is only registered in the deployed version.

**Reply by CGNonofr** [reply]
I don't really know, I think service-worker are used by VSCode as a "proxy" from virtual urls to data
